public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx

```diff
@@ -80,12 +100,28 @@ export function panelMenuBehavior(menu: VizPanelMenu) {
       shortcut: 'p s',
     });
 
+    moreSubMenu.push({
+      text: t('panel.header-menu.duplicate', `Duplicate`),
+      onClick: () => {
+        DashboardInteractions.panelMenuItemClicked('duplicate');
+        dashboard.duplicatePanel(panel);
+      },
+      shortcut: 'p d',
+    });
+
+    moreSubMenu.push({
+      text: t('panel.header-menu.copy', `Copy`),
+      onClick: () => {
+        DashboardInteractions.panelMenuItemClicked('copy');
+        dashboard.copyPanel(panel);
+      },
+    });
+
     if (panel.parent instanceof LibraryVizPanel) {
       // TODO: Implement lib panel unlinking
     } else {
       moreSubMenu.push({
         text: t('panel.header-menu.create-library-panel', `Create library panel`),
-        iconClassName: 'share-alt',
         onClick: () => {
           DashboardInteractions.panelMenuItemClicked('createLibraryPanel');
           dashboard.showModal(
```

Comment:
I've noticed that this one doesn't really work as expected, so I should probably update this one too so it matches the dashboard core behavior

Reply:
Yeah, I've noticed this as well, you have used wrong tab id. I've pushed an update.